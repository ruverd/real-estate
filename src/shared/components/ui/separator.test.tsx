import { render } from "@testing-library/react";

import { Separator } from "./separator";

describe("Separator", () => {
  it("should renders horizontal separator by default", () => {
    const { container } = render(<Separator />);
    
    const separator = container.querySelector('[data-slot="separator-root"]');

    expect(separator).toBeInTheDocument();
    expect(separator).toHaveAttribute("data-orientation", "horizontal");
  });

  it("should renders vertical separator when orientation is set to vertical", () => {
    const { container } = render(<Separator orientation="vertical" />);
    
    const separator = container.querySelector('[data-slot="separator-root"]');

    expect(separator).toBeInTheDocument();
    expect(separator).toHaveAttribute("data-orientation", "vertical");
  });

  it("should applies custom className", () => {
    const { container } = render(<Separator className="custom-class" />);
    
    const separator = container.querySelector('[data-slot="separator-root"]');

    expect(separator).toHaveClass("custom-class");
  });

  it("should match snapshot", () => {
    const { container } = render(<Separator />);

    expect(container).toMatchSnapshot();
  });
});
