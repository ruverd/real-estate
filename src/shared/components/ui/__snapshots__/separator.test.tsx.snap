// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > should match snapshot 1`] = `
<div>
  <div
    class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px"
    data-orientation="horizontal"
    data-slot="separator-root"
    role="none"
  />
</div>
`;
